src/hotspot/cpu/aarch64/compressedKlass_aarch64.cpp

```diff
@@ -58,6 +61,19 @@ static char* reserve_at_eor_compatible_address(size_t size, bool aslr) {
       0x7ffc, 0x7ffe, 0x7fff
   };
   static constexpr int num_immediates = sizeof(immediates) / sizeof(immediates[0]);
+  if (aslr) {
+    // With aslr enabled, call os::init_random so that we get a different start_index
+    // in each run.
+#if INCLUDE_CDS
+    if (!CDSConfig::is_dumping_static_archive()) {
+      // To ensure we have a deterministic static archive on aarch64 platform, call
+      // os::init_random when not dumping static archive.
```

Comment:
Does this code conflict with the call to `os::init_random(0x12345678)` in [ VM_PopulateDumpSharedSpace::doit()](https://github.com/openjdk/jdk/blob/2063bb8ffabd6096f547ec6da979cfcf68a56ba3/src/hotspot/share/cds/metaspaceShared.cpp#L499-L517)?

I think it's better to remove the call to os::init_random(), and change the following call to

```
const int start_index = aslr ? os::next_random(int)os::javaTimeNanos()) : 0;
``` 

That way we can avoid modifying a global state that may affect other VM components.

I also created https://bugs.openjdk.org/browse/JDK-8323900 to avoid calling `os::init_random(0x12345678)`

Reply:
> Does this code conflict with the call to `os::init_random(0x12345678)` in [ VM_PopulateDumpSharedSpace::doit()](https://github.com/openjdk/jdk/blob/2063bb8ffabd6096f547ec6da979cfcf68a56ba3/src/hotspot/share/cds/metaspaceShared.cpp#L499-L517)?

Not sure. Without the above static dump check, the `DeterministicDump.java` test failed.

> 
> I think it's better to remove the call to os::init_random(), and change the following call to
> 
> ```
> const int start_index = aslr ? os::next_random(int)os::javaTimeNanos()) : 0;
> ```

I tried the above change and it works.

> 
> That way we can avoid modifying a global state that may affect other VM components.
> 
> I also created https://bugs.openjdk.org/browse/JDK-8323900 to avoid calling `os::init_random(0x12345678)`

Thanks.